src/yunohost/permission.py

```diff
@@ -0,0 +1,510 @@
+                 continue
+            if auth.update('cn=%s,ou=permission' % permission_name, default_permission):
+                logger.success(m18n.n('permission_updated', permission=per, app=a))
+            else:
+                raise YunohostError('permission_update_failed')
+
+    permission_sync_to_user(auth)
+
+    for a in app:
+        permission_name = 'main.' + a
+        result = auth.search('ou=permission,dc=yunohost,dc=org',
+                             filter='cn=' + permission_name, attrs=['inheritPermission'])
+        if result:
+            allowed_users = result[0]['inheritPermission']
+            new_user_list = ','.join(allowed_users)
+            hook_callback('post_app_removeaccess', args=[app, new_user_list])
+
+    return user_permission_list(auth, app, permission)
+
+
+@is_unit_operation(['permission','app'])
+def permission_add(operation_logger, auth, app, permission, urls=None, default_allow=True, sync_perm=True):
+    """
+    Create a new permission for a specific application
+
+    Keyword argument:
+        app        -- an application OR sftp, xmpp (metronome), mail
+        permission -- name of the permission ("main" by default)
+        urls       -- list of urls to specify for the permission
+
+    """
+    from yunohost.domain import _normalize_domain_path
+
+    # Validate uniqueness of permission in LDAP
+    permission_name = str(permission + '.' + app) # str(...) Fix encoding issue
+    conflict = auth.get_conflict({
+        'cn': permission_name
+    }, base_dn='ou=permission,dc=yunohost,dc=org')
+    if conflict:
+        raise YunohostError('permission_already_exist', permission=permission, app=app)
+
+    # Get random GID
+    all_gid = {x.gr_gid for x in grp.getgrall()}
+
+    uid_guid_found = False
+    while not uid_guid_found:
+        gid = str(random.randint(200, 99999))
+        uid_guid_found = gid not in all_gid
+
+    attr_dict = {
+        'objectClass': ['top', 'permissionYnh', 'posixGroup'],
+        'cn': permission_name,
+        'gidNumber': gid,
+    }
+    if default_allow:
+        attr_dict['groupPermission'] = 'cn=all_users,ou=groups,dc=yunohost,dc=org'
+
+    if urls:
+        attr_dict['URL'] = []
+        for url in urls:
+            domain = url[:url.index('/')]
+            path = url[url.index('/'):]
+            domain, path = _normalize_domain_path(domain, path)
+            attr_dict['URL'].append(domain + path)
+
+    operation_logger.start()
+    if auth.add('cn=%s,ou=permission' % permission_name, attr_dict):
+        if sync_perm:
+            permission_sync_to_user(auth)
+        logger.success(m18n.n('permission_created', permission=permission, app=app))
+        return user_permission_list(auth, app, permission)
+
+    raise YunohostError('permission_creation_failed')
+
+
+@is_unit_operation(['permission','app'])
+def permission_update(operation_logger, auth, app, permission, add_url=None, remove_url=None, sync_perm=True):
```

Comment:
It seems to me that this `update` action ain't available from the actionmap and should be ? Or am I missing something ?

Reply:
Yes because it's internal. So no user should use this function.